object_detection/jobs/pets-training.yaml

```diff
@@ -1,77 +1,61 @@
 ---
-apiVersion: kubeflow.org/v1alpha1
+apiVersion: kubeflow.org/v1alpha2
 kind: TFJob
 metadata:
   name: pets-training
 spec:
-  replicaSpecs:
-  - replicas: 1
-    template:
-      spec:
-        containers:
-        - image: lcastell/pets_object_detection
-          imagePullPolicy: IfNotPresent
-          name: tensorflow
-          workingDir: /models
-          command:
-            - "python"
-            - "research/object_detection/train.py"
-          args:
-            - "--logtostderr"
-            - "--pipeline_config_path=/pets_data/faster_rcnn_resnet101_pets.config"
-            - "--train_dir=/pets_data/train"
-          volumeMounts:
-          - mountPath: "/pets_data"
-            name: pets-data
-        volumes:
-        - name: pets-data
-          persistentVolumeClaim:
-            claimName: pets-data-claim
-        restartPolicy: OnFailure
-    tfReplicaType: MASTER
-  - replicas: 1
-    template:
-      spec:
-        containers:
-        - image: lcastell/pets_object_detection
-          imagePullPolicy: IfNotPresent
-          name: tensorflow
-          command:
-            - "python"
-            - "research/object_detection/train.py"
-          args:
-            - "--logtostderr"
-            - "--pipeline_config_path=/pets_data/faster_rcnn_resnet101_pets.config"
-            - "--train_dir=/pets_data/train"
-          volumeMounts:
-          - mountPath: "/pets_data"
-            name: pets-data
-        volumes:
-        - name: pets-data
-          persistentVolumeClaim:
-            claimName: pets-data-claim
-        restartPolicy: OnFailure
-    tfReplicaType: WORKER
-  - replicas: 1
-    template:
-      spec:
-        containers:
-        - image: lcastell/pets_object_detection
-          imagePullPolicy: IfNotPresent
-          name: tensorflow
-          command:
-            - "python"
-            - "research/object_detection/train.py"
-          args:
-            - "--logtostderr"
-            - "--pipeline_config_path=/pets_data/faster_rcnn_resnet101_pets.config"
-            - "--train_dir=/pets_data/train"
-          volumeMounts:
-          - mountPath: "/pets_data"
-            name: pets-data
-        volumes:
-        - name: pets-data
-          persistentVolumeClaim:
-            claimName: pets-data-claim
-        restartPolicy: OnFailure
-    tfReplicaType: PS
\ No newline at end of file
+  tfReplicaSpecs:
+    PS:
+      replicas: 1
+      restartPolicy: OnFailure
+      template:
+        spec:
+          containers:
+          - name: tensorflow
+            image: lcastell/pets_object_detection
+            resources:
+              limits:
+                nvidia.com/gpu: 1
```

Comment:
Question, if your cluster does not have GPU, will this make the job fail?

Reply:
I agree with @jlewi I have started to look into parametrizing the manifests into a ksonnet app, and address this issue too.
I will see if I can push something by next weekend

Reply:
Oh yes, I forget about this part, I gonna remove the GPU part.

and Agree with that ksonnet would be much better